tests/harness/cases/messages.proto

```diff
@@ -28,3 +32,18 @@ message MessageSkip { TestMsg val = 1 [(validate.rules).message.skip = true];}
 message MessageRequired { TestMsg val = 1 [(validate.rules).message.required = true]; }
 
 message MessageWith3dInside {}
+
+message ComplexTestMsg {
+    string                     const  = 1 [(validate.rules).string.const = "abcd"];
+    ComplexTestMsg             nested = 2;
+    int32                      int_const = 3 [(validate.rules).int32.const = 5];
+    bool                       bool_const = 4 [(validate.rules).bool.const = false];
+    google.protobuf.FloatValue float_val = 5 [(validate.rules).float.gt = 0];
+    google.protobuf.Duration   dur_val = 6 [(validate.rules).duration.lt = {seconds: 17}, (validate.rules).duration.required = true];
+    google.protobuf.Timestamp  ts_val = 7 [(validate.rules).timestamp.gt = {seconds: 7}];
+    TestMsg                    another = 8;
+    float                      float_const = 9 [(validate.rules).float.lt = 8];
+    double                     double_in = 10 [(validate.rules).double = {in: [456.789, 123]}];
+}
```

Comment:
Let's move this to a separate proto and name it something like "kitchensink" or what have you. For the most part, the test messages should be simple so that we can isolate the behavior of the various rules, but there is certainly value in testing their interaction like this. It should just happen in its own isolated space to reduce noise.